src/main/java/org/jenkinsci/plugins/workflow/cps/CpsFlowExecution.java

```diff
@@ -384,8 +385,7 @@ public void setResumeBlocked(boolean resumeBlocked) {
     }
 
     /** accumulated time in ns of a given {@link TimingKind#name}; {@link String} key for pretty XStream form */
-    @GuardedBy("this")
-    @CheckForNull Map<String, Long> timings;
+    @NonNull Map<String, Long> timings = new ConcurrentHashMap<>();
```

Comment:
We just got rid of concurrent persistent fields in #518, let us not now reintroduce one! :grin: (see that PR for how to run the relevant tests in Java 17)

Reply:
Actually, the test command passes for me. Probably related to https://github.com/jenkinsci/workflow-cps-plugin/pull/519/files#diff-1c6ff2489c8f8b9b41176212ef2e09f3e26ee2967a9ed4a8b483710026d2cbe7R1640

Reply:
Oh I see.

Reply:
I'm not sure whether that code actually changes the serialized type. It might just be that XStream has a special converter for for `ConcurrentHashMap` but not `ConcurrentSkipListMap`, e.g. https://github.com/x-stream/xstream/blob/2d2f5e9b407a4b0877a4e8cd9f3b18d019fe4b0d/xstream/src/java/com/thoughtworks/xstream/converters/collections/MapConverter.java#L71-L75.

Reply:
I think I am seeing this very problem where a test is deserializing an old build, but it is failing on the `close` method due to `timings` being null